larray/inout/xw_reporting.py

```diff
@@ -0,0 +1,589 @@
+            (see :py:obj:`~ExcelGraphItem.set_default_size`). Defaults to None.
+        height : int, optional
+            height of the title item. The current default value is used if None
+            (see :py:obj:`~ExcelGraphItem.set_default_size`). Defaults to None.
+
+        Examples
+        --------
+        >>> demo = load_example_data('demo')
+        >>> templates_dir = get_example_templates_dir()
+        >>> report = ExcelReport(templates_dir)
+
+        >>> report.set_sheet('Belgium')
+
+        Using default 'template', 'width' and 'height'
+
+        >>> report.add_graph('Population', demo.pop['Belgium'], template='Line')
+
+        Specifying the 'width' and 'height' values
+
+        >>> report.add_graph('Births', demo.births['Belgium'], template='Line', width=450, height=250)
+
+        Setting a default template
+
+        >>> report.set_graph_template('Line_Marker')
+        >>> report.add_graph('Deaths', demo.deaths['Belgium'])
+
+        Dumping the report Excel file
+
+        >>> # do not forget to call 'to_excel' to create the report file
+        >>> report.to_excel('Demography_Report.xlsx')
+        """
+        pass
+
+    def to_excel(self, filepath, data_sheet_name='__data__', overwrite=True):
+        r"""
+        Generate the report Excel file.
+
+        Parameters
+        ----------
+        filepath : string
+            Path of the report file for the dump.
+        data_sheet_name : string, optional
+            name of the Excel sheet where all data associated with items is dumped.
+            Defaults to '__data__'.
+        overwrite : bool, optional
+            whether or not to overwrite an existing report file.
+            Defaults to True.
+
+        Examples
+        --------
+        >>> demo = load_example_data('demo')
+        >>> templates_dir = get_example_templates_dir()
+        >>> report = ExcelReport(templates_dir)
+        >>> report.set_graph_template('Line_Marker')
+
+        >>> for c in demo.country:
+        ...     report.set_sheet(c)
+        ...     report.add_graph('Population', demo.pop[c])
+        ...     report.add_graph('Births', demo.births[c])
+        ...     report.add_graph('Deaths', demo.deaths[c])
+
+        Basic usage
+
+        >>> report.to_excel('Demography_Report.xlsx')
+
+        Alternative data sheet name
+
+        >>> report.to_excel('Demography_Report.xlsx', data_sheet_name='Data Tables') # doctest: +SKIP
+
+        Check if ouput file already exists
+
+        >>> report.to_excel('Demography_Report.xlsx', overwrite=False) # doctest: +SKIP
+        Traceback (most recent call last):
+        ...
+        ValueError: Sheet named 'Belgium' already present in workbook
+        """
+        pass
+
+
+if xw is not None:
+    from xlwings.constants import LegendPosition, HAlign, VAlign, ChartType, RowCol, Direction
+    from larray.inout.xw_excel import open_excel
+
+    def create_instance(app_id):
```

Comment:
this special create_instance code should be used in/moved to open_excel (unsure if this is reliable enough to be the default behavior or not)